src/reader/buffered_reader.rs

```diff
@@ -652,17 +660,63 @@ mod test {
     use super::*;
     use crate::reader::test::check;
 
-    fn input_from_bytes(bytes: &[u8]) -> BufferedReader<&[u8]> {
-        BufferedReader(bytes)
+    fn input_from_str(s: &str) -> BufferedReader<&[u8]> {
+        BufferedReader(s.as_bytes())
     }
 
     fn reader_from_str(s: &str) -> Reader<BufferedReader<&[u8]>> {
         Reader::from_reader_internal(BufferedReader(s.as_bytes()))
     }
 
     #[allow(dead_code)]
-    fn reader_from_bytes(s: &[u8]) -> Reader<BufferedReader<&[u8]>> {
-        Reader::from_reader_internal(BufferedReader(s))
+    fn reader_from_bytes(bytes: &[u8]) -> Reader<BufferedReader<&[u8]>> {
+        Reader::from_reader_internal(BufferedReader(bytes))
+    }
+
+    #[cfg(feature = "encoding")]
```

Comment:
Moving these because it doesn't make sense to check these for both when `SliceReader` can't work with encoded bytes.